tony-core/src/main/java/com/linkedin/tony/Connection.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.tony;
+
+interface Connection extends AutoCloseable {
```

Comment:
This object structure diagram design looks weird to me.

It is more natural to have a BaseClass named Connection and then have a derived class named PortReusableConnection.

Like Array vs MutableArray in other languages.

Reply:
thanks, changed to abstract class.